contrib/automation_tests/test_cases/symbols_tab.py

```diff
@@ -18,21 +23,229 @@
+        pass
+
+    def _check_and_update_duration(self, storage_key: str, total_duration: float, expected_difference: float):
+        """
+        Compare the current duration with the previous run, and fail if it exceeds the defined bounds.
+        :param storage_key: Unique key to store the duration within the test suite, used to persist the data across
+            multiple tests (uses `self.suite.shared_data`)
+        :param total_duration: The duration of the current run
+        :param expected_difference: Expected difference in seconds. If negative, the current run must be at least
+            this much *faster* than the previous run. If positive, the current run must be at least this much *slower*.
+            Violating those conditions will result in test failure.
+        """
+        last_duration = self.suite.shared_data.get(storage_key, None)
+        self.suite.shared_data[storage_key] = total_duration
+
+        if expected_difference is not None and last_duration is not None:
+            if expected_difference < 0:
+                self.expect_true(total_duration <= last_duration + expected_difference,
+                                 "Expected symbol loading time to be at least {diff:.2f}s faster than the last run. "
+                                 "Last run duration: {last:.2f}s, current run duration: {cur:.2f}s".format(
+                                     diff=-expected_difference,
+                                     last=last_duration,
+                                     cur=total_duration))
+            else:
+                self.expect_true(total_duration >= last_duration + expected_difference,
+                                 "Expected symbol loading time to be at least {diff:.2f}s slower than the last run. "
+                                 "Last run duration: {last:.2f}s, current run duration: {cur:.2f}s".format(
+                                     diff=expected_difference,
+                                     last=last_duration,
+                                     cur=total_duration))
+
+
+ModulesLoadingResult = namedtuple("LoadingResult", ["time", "errors"])
+
+
+class ClearSymbolCache(E2ETestCase):
+    """
+    Clear the symbol cache and verify no symbol files remain.
+    """
+
+    def _execute(self):
+        logging.info("Clearing symbol cache at {location}".format(location=CACHE_LOCATION))
+        for file in os.listdir(CACHE_LOCATION):
+            full_path = os.path.join(CACHE_LOCATION, file)
+            if os.path.isfile(full_path):
+                os.unlink(full_path)
+        self.expect_true(not os.listdir(CACHE_LOCATION), 'Cache is empty')
+
+
+class LoadAllSymbolsAndVerifyCache(E2ETestCase, DurationDiffMixin):
+    """
+    Loads all symbol files at once and checks if all of them exist in the cache. In addition, this test measures the
+    total duration of symbol loading (estimated), and can fail if the duration differs from the previous run by
```

Comment:
> ...can fail if the duration differs from the previous run...

I guess this might become a problem when running on the testing infrastructure. But let's see  - if the bounds are generous enough and it never shows up this should be fine. Same for LoadSymbols below.

Reply:
Yes, I'm worried about this as well. But I don't really see a good alternative to verify if the application is actually using the cached symbols. Maybe we can change this with the new symbol loading functionality if the UI actually gives us an indication of whether (!) symbols are cached or will be downloaded?